examples/99-advanced/openfoam-cooling.py

```diff
@@ -0,0 +1,173 @@
+
+# Plot it
+pl = pv.Plotter()
+pl.enable_ssao(radius=0.01)
+pl.add_mesh(z_slice, scalars='U', lighting=False, scalar_bar_args={'title': 'Velocity'})
+pl.add_mesh(structure, color='w', smooth_shading=True, split_sharp_edges=True)
+pl.camera_position = 'xy'
+pl.camera.roll = 90
+pl.enable_anti_aliasing('fxaa')
+pl.show()
+
+
+###############################################################################
+# Plot Air Temperature
+# ~~~~~~~~~~~~~~~~~~~~
+# Let's also plot the temperature of the air. This time, let's also plot the
+# temperature of the components.
+
+pl = pv.Plotter()
+pl.enable_ssao(radius=0.01)
+pl.add_mesh(
+    z_slice, scalars='T', lighting=False, scalar_bar_args={'title': 'Temperature'}, cmap='reds'
+)
+pl.add_mesh(
+    structure,
+    scalars='T',
+    smooth_shading=True,
+    split_sharp_edges=True,
+    cmap='reds',
+    show_scalar_bar=False,
+)
+pl.camera_position = 'xy'
+pl.camera.roll = 90
+pl.enable_anti_aliasing('fxaa')
+pl.show()
+
+
+###############################################################################
+# Plot Steamlines - Flow Velocity
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Now, let's plot the steamlines of this dataset so we can see how the air is
+# flowing through the case.
+#
+# Generate streamlines using :func:`streamlines() <pyvista.DataSetFilters.streamlines>`.
+
+# Have our streamlines start from the regular openings of the case.
+lines = []
+z_rng = np.linspace(0, 0.03, 5)
+for x in np.linspace(0.045, 0.105, 7, endpoint=True):
+    lines.extend([air.streamlines(start_position=[x, 0.2, z], max_time=2.0) for z in z_rng])
+lines = pv.merge(lines)
+
+# Plot
+pl = pv.Plotter()
+pl.enable_ssao(radius=0.01)
+pl.add_mesh(lines, line_width=2, scalars='T', cmap='reds', scalar_bar_args={'title': 'Temperature'})
+pl.add_mesh(
+    structure,
+    scalars='T',
+    smooth_shading=True,
+    split_sharp_edges=True,
+    cmap='reds',
+    show_scalar_bar=False,
+)
+pl.camera_position = 'xy'
+pl.camera.roll = 90
+pl.enable_anti_aliasing('fxaa')  # also try 'ssaa'
+pl.show()
+
+
+###############################################################################
+# Volumetric Plot - Visualize High Temperatures
+# ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+# Show a 3D plot of areas of temperature.
+#
+# For this example, we will first interpolate the results from the
+# :class:`pyvista.UnstructuredGrid` onto a :class:`pyvista.UniformGrid` using
+# :func:`interpolate() <pyvista.DataSetFilters.interpolate>`. This is so we can
+# visualize it using :func:`add_volume() <pyvista.Plotter.add_volume>`
+
+bounds = np.array(air.bounds) * 1.2
+origin = (bounds[0], bounds[2], bounds[4])
+spacing = (0.003, 0.003, 0.003)
+dimensions = (
+    int((bounds[1] - bounds[0]) // spacing[0] + 2),
+    int((bounds[3] - bounds[2]) // spacing[1] + 2),
+    int((bounds[5] - bounds[4]) // spacing[2] + 2),
+)
+grid = pv.UniformGrid(dimensions=dimensions, spacing=spacing, origin=origin)
+grid = grid.interpolate(air, radius=0.005)
```

Comment:
Similar comment to tubes example, for data with cells it is more faithful to the computational approach to use `probe` or `sample`.  This is particularly important for meshes with a wide distribution of element sizes common in CFD.

Reply:
Resolved in 2a78fa8aede41f258650f2a7514d295763d43ac8.